### PR TITLE
http proxy support

### PR DESCRIPTION
**Pull request by dakale**
Replaces typed-rest-client with http-client and upgrades tool-cache to versions that support http(s)_proxy / no_proxy from the environment

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -24,3 +24,34 @@ jobs:
...
+      http_proxy: http://localhost:3128
+      https_proxy: http://localhost:3128
+    steps:
+    - name: Block non proxied traffic
+      run: |
+        echo "127.0.0.0 registry.npm.js nodejs.org github.com api.github.com download.java.net static.azul.com" | sudo tee -a /etc/hosts
+    - name: Checkout
+      uses: actions/checkout@master
+
+    - name: Set Node.js 10.x
```
Comment (ericsciple) [inline .github/workflows/workflow.yml]: runner uses node 12

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -24,3 +24,34 @@ jobs:
...
+        options: '--health-cmd "exit 0" --health-interval 3s'
+    env:
+      http_proxy: http://localhost:3128
+      https_proxy: http://localhost:3128
+    steps:
+    - name: Block non proxied traffic
+      run: |
+        echo "127.0.0.0 registry.npm.js nodejs.org github.com api.github.com download.java.net static.azul.com" | sudo tee -a /etc/hosts
+    - name: Checkout
+      uses: actions/checkout@master
```
Comment (ericsciple) [inline .github/workflows/workflow.yml]: cehckout v2

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -24,3 +24,34 @@ jobs:
...
+
+    - name: Set Node.js 10.x
+      uses: actions/setup-node@master
```
Comment (ericsciple) [inline .github/workflows/workflow.yml]: setup node v1

**Comment by dakale** [comment]
@konradpabjan Are you the right person to review this?

Also, @ericsciple is going to follow up on this (in a separate pr) to add some e2e tests that test package install/restore, etc, scenarios after running the java setup

File: src/installer.ts [hunk 4]
```diff
@@ -3,11 +3,11 @@ let tempDirectory = process.env['RUNNER_TEMP'] || '';
 import * as core from '@actions/core';
 import * as io from '@actions/io';
 import * as exec from '@actions/exec';
+import * as httpm from '@actions/http-client';
```
Comment (ericsciple) [inline src/installer.ts]: consider add retries and error checking (or switch to getJson which does error checking)

**Comment by ericsciple** [comment]
@konradpabjan can you merge, david is on call i'll follow up with the minor enhancements (switch to node 12, checkout v2, node v1, etc)

**Pull request merged**